google/cloud/security/common/util/metadata_server.py

```
@@ -20,29 +20,87 @@
+        headers: A dictionary reprsenting key-value pairs of headers.
+
+    Returns:
+        The httplib.HTTPResponse object.
 
+    Raises:
+        MetadataHttpError: When we can't reach the requested host.
+    """
+    http_client = _obtain_http_client()
+
+    try:
+        return http_client.request(path, method, headers)
+    except socket.error as e:
+        raise MetadataHttpError(e)
+    finally:
+      http_client.close()
 
 def can_reach_metadata_server():
     """Determine if we can reach the metadata server.
 
     Returns:
         True if metadata server can be reached, False otherwise.
     """
-    conn = httplib.HTTPConnection('metadata.google.internal',
-                                  timeout=METADATA_SERVER_CONN_TIMEOUT)
-    can_reach_md_server = False
+    path = '/computeMetadata/v1/instance/id'
+    response = None
 
     try:
-        conn.request('GET', '/computeMetadata/v1/instance/id',
-                     None, {'Metadata-Flavor': 'Google'})
-        res = conn.getresponse()
-        can_reach_md_server = (res and res.status == 200)
-    except socket.error:
-        # Don't care about showing that an error happened.
+        response = _issue_http_request(
+            path, HTTP_GET, REQUIRED_METADATA_HEADER)
+    except MetadataHttpError as e:
         pass
 
-    finally:
-        conn.close()
+    return (response and response.status == HTTP_SUCCESS)
 
-    return can_reach_md_server
+def get_value_for_attribute(attribute):
+    """For a given key return the value.
+
+    Args:
+        attribute: a String representing the key.
+
+    Returns:
+        The value of the request.
+    """
+    path = '/computeMetadata/v1/instance/attributes/'
+    path = path + attribute
+    http_response = ''
+
+    try:
+        http_response = _issue_http_request(
+            path, HTTP_GET, REQUIRED_METADATA_HEADER)
+        return http_response.read()
+    except MetadataHttpError:
+        return http_response
```

should we return None or raise an error instead of an empty string?